packages/ember-self-focused/addon/services/focus-manager.js

```diff
@@ -96,8 +96,12 @@ export default Service.extend({
       const scrollY = window.pageYOffset;
       node.setAttribute('tabindex', '-1');
       node.focus();
-      // after setting focus, scroll back to the place where the user was previously
-      window.scrollTo(scrollX, scrollY);
+      // After setting focus, scroll back to the place where the user was previously.
+      // We temporarily need to use a setTimeout because in Safari 12 there is an issue
+      // where the scroll that happens internally from focusing seems to clobber our
+      // scrollTo call. Calling setTimeout with a timeout of 0 ensures that scrollTo
+      // is only executed after the current call stack is empty
+      setTimeout(() => window.scrollTo(scrollX, scrollY), 0);
```

Comment:
Last time @yuclinandreas and I tried out this didn't work and we had to use

```
document.body.scrollTop = scrollTop;
document.body.scrollLeft = scrollLeft;
```

Reply:
Could you clarify what you mean? Do you mean that you and @yuclinandreas tried to use setTimeout and it didn't work? Which browser didn't it work on? I tried it locally and didn't have any problems. I'm trying to understand what issue using scrollTop and scrollLeft fixes

Reply:
^ Above conversation was resolved on Slack. Both approaches work, but document.body.scrollTop and document.body.scrollLeft doesn't require setTimeout, which is the better bet because using setTimeout is generally frowned upon since it doesn't play nice with components/ember run loop